services/requester/requester.go

```diff
@@ -159,12 +162,36 @@ func (e *EVM) SendRawTransaction(ctx context.Context, data []byte) (common.Hash,
 		return common.Hash{}, err
 	}
 
-	err := models.ValidateTransaction(tx)
-	if err != nil {
+	if err := models.ValidateTransaction(tx); err != nil {
 		return common.Hash{}, err
 	}
 
-	// todo do further validation
+	head := &types.Header{
+		Number:   big.NewInt(20_182_324),
+		Time:     uint64(time.Now().Unix()),
+		GasLimit: 30_000_000,
+	}
+	chainConfig := emulator.DefaultChainConfig
+	chainConfig.ChainID = e.config.EVMNetworkID
+	signer := types.MakeSigner(
+		chainConfig,
+		head.Number,
+		head.Time,
+	)
+	opts := &txpool.ValidationOptions{
+		Config: chainConfig,
+		Accept: 0 |
+			1<<types.LegacyTxType |
+			1<<types.AccessListTxType |
+			1<<types.DynamicFeeTxType |
+			1<<types.BlobTxType,
+		MaxSize: models.TxMaxSize,
+		MinTip:  new(big.Int),
+	}
+
+	if err := models.ValidateConsensusRules(tx, head, signer, opts); err != nil {
```

Comment:
can we call this inside the `ValidateTransaction`

Reply:
We could add this logic to the existing `ValidateTransaction` method. I just introduced a different method, to be able to test it in isolation.

Reply:
Updated in https://github.com/onflow/flow-evm-gateway/pull/323/commits/4ad00ea8d95b9343b7d61ef024a375c80d82b99b